packages/realm/src/app-services/ApiKeyAuthClient.ts

```diff
@@ -30,7 +31,7 @@ export type ApiKey = {
   /**
    * The secret part of the key.
    */
-  key: string;
+  key?: string;
```

Comment:
the case would be when the key is received via `fetch(id)` but I'm not certain if it was treated in another way before

Reply:
didn't know you could do it that way, cool stuff!

Reply:
hmm the linter complains if I remove the optional property in TS for the `Create()` method where we always return a string rather than undefined. Can't bypass this either with the `!` operator since the linter complains there as well